spec/spec_helper.rb

```diff
@@ -62,6 +62,10 @@ def handle_current_dir_change
   ensure
     RSpec::Core::Metadata.instance_variable_set(:@relative_path_regex, nil)
   end
+
+  def with_an_expected_warning
+    with_isolated_stderr { yield }
+  end
```

Comment:
How do you feel about making this expect an error, e.g. at minimum
```suggestion
  def with_an_expected_warning(warning = //)
    expect { yield }.to output(warning).to_stderr
  end
```
This would be an actually expected warning then, optionally we could assert on the specifics?

Reply:
Didn't work as is, had to rework as
```ruby
return_value = nil
expect { return_value = yield }.to output(warning).to_stderr
return_value
```
and it complicates `subject_value_for` in memoized_helpers_spec.rb, as some usages do output and some don't.
The potential to catch unexpected warnings may not overweight the added complexity.